pkg/certificateprovider/azurekeyvault/provider.go

```diff
@@ -0,0 +1,207 @@
+func GetCertificates(ctx context.Context, attrib map[string]string) ([]types.Certificate, error) {
+	keyvaultUri := types.GetKeyVaultUri(attrib)
+	cloudName := types.GetCloudName(attrib)
+	tenantID := types.GetTenantID(attrib)
+	workloadIdentityClientID := types.GetClientID(attrib)
+
+	if keyvaultUri == "" {
+		return nil, fmt.Errorf("keyvaultUri is not set")
+	}
+	if tenantID == "" {
+		return nil, fmt.Errorf("tenantID is not set")
+	}
+	if workloadIdentityClientID == "" {
+		return nil, fmt.Errorf("clientID is not set")
+	}
+
+	azureCloudEnv, err := parseAzureEnvironment(cloudName)
+	if err != nil {
+		return nil, fmt.Errorf("cloudName %s is not valid, error: %w", cloudName, err)
+	}
+
+	// 1. cleaning up keyvault objects definition
+	certificatesStrings := types.GetCertificates(attrib)
+	if certificatesStrings == "" {
+		return nil, fmt.Errorf("certificates is not set")
+	}
+
+	common.LogDebug("certificates string defined in ratify certStore class, certificates %v", certificatesStrings)
+
+	objects, err := types.GetCertificatesArray(certificatesStrings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to yaml unmarshal objects, error: %w", err)
+	}
+	common.LogDebug("unmarshaled objects yaml, objectsArray %v", objects.Array)
+
+	keyVaultCerts := []types.KeyVaultCertificate{}
+	for i, object := range objects.Array {
+		var keyVaultCert types.KeyVaultCertificate
+		if err = yaml.Unmarshal([]byte(object), &keyVaultCert); err != nil {
+			return nil, fmt.Errorf("unmarshal failed for keyVaultCerts at index %d, error: %w", i, err)
+		}
+		// remove whitespace from all fields in keyVaultCert
+		formatKeyVaultCertificate(&keyVaultCert)
+
+		keyVaultCerts = append(keyVaultCerts, keyVaultCert)
+	}
+
+	common.LogDebug("unmarshaled %v key vault objects, keyVaultObjects: %v", len(keyVaultCerts), keyVaultCerts)
+
+	if len(keyVaultCerts) == 0 {
+		return nil, fmt.Errorf("no keyvault certificate configured")
+	}
+
+	common.LogDebug("vaultURI %s", keyvaultUri)
+
+	kvClient, err := initializeKvClient(ctx, azureCloudEnv.KeyVaultEndpoint, tenantID, workloadIdentityClientID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get keyvault client, error: %w", err)
+	}
+
+	certs := []types.Certificate{}
+	for _, keyVaultCert := range keyVaultCerts {
+		common.LogDebug("fetching object from key vault, certName %v,  keyvault %v", keyVaultCert.CertificateName, keyvaultUri)
+
+		// fetch the object from Key Vault
+		result, err := getCertificate(ctx, kvClient, keyvaultUri, keyVaultCert)
+		if err != nil {
+			return nil, err
+		}
+
+		for idx := range result {
+			r := result[idx]
+			objectContent := []byte(r.content)
+
+			cert := types.Certificate{
+				CertificateName: keyVaultCert.CertificateName,
+				Content:         objectContent,
+				Version:         r.version,
+			}
+
+			certs = append(certs, cert)
+			common.LogDebug("added certificates %v to response", cert.CertificateName)
+		}
```

Comment:
After `getCertificate` is modified to return a single cert instead of slice, this for loop can be removed